src/utils/json.ts

```diff
@@ -0,0 +1,48 @@
+// Regex to clean up added $refId & $ref properties, added by Realm.JsonSerializationReplacer
+// TODO: Investigate better solution.
+const $REF_MATCHER = /\s*\"\$ref[Id]*\" *: *\".*\"(,|(?=\s+\}))/;
+
+const DEFAULT_POSTFIX = ' (...)';
+const INDENTATION = 2;
+
+type SafeJsonOptions = {
+  pretty?: boolean;
+  cleanupRefs?: boolean;
+  maxLength?: number;
+  postFix?: string;
+};
+
+export const asSafeJsonString = (
+  value: unknown,
+  options?: SafeJsonOptions,
+): string => {
+  let json: string;
+  const indentation = options?.pretty ? INDENTATION : undefined;
+
+  try {
+    json = JSON.stringify(value, null, indentation);
+  } catch {
+    try {
+      json = JSON.stringify(
+        value,
+        Realm.JsonSerializationReplacer,
+        indentation,
+      );
+    } catch (err) {
+      json = err.message ? err.message : err;
+    }
+  }
+
+  if (options) {
+    if (options.cleanupRefs) {
+      json = json.replace($REF_MATCHER, '');
```

Comment:
This will look a bit broken in the output won't it?

Reply:
For what I tested, it looked as intended. There might be edge-cases, where the solution mentioned in the previous comment would avoid any/all edge-cases.